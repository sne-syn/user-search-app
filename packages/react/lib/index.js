export { default as Avatar } from './atoms/Avatar/Avatar.js';
export { default as Button } from './atoms/Button/Button.js';
export { default as CardSubtitle } from './atoms/CardSubtitle/CardSubtitle.js';
export { default as CardTitle } from './atoms/CardTitle/CardTitle.js';
export { default as Icon } from './atoms/Icon/Icon.js';
export { default as Input } from './atoms/Input/Input.js';
export { default as Logo } from './atoms/Logo/Logo.js';
export { default as DateInfo } from './atoms/DateInfo/DateInfo.js';
export { default as DefinitionList } from './atoms/DefinitionList/DefinitionList.js';
export { default as ErrorMessage } from './atoms/ErrorMessage/ErrorMessage.js';
export { default as TextBlock } from './atoms/TextBlock/TextBlock.js';
export { default as ThemeToggler } from './atoms/ThemeToggler/ThemeToggler.js';
export { default as VisuallyHidden } from './atoms/VisuallyHidden/VisuallyHidden.js';
export { default as Wrapper } from './atoms/Wrapper/Wrapper.js';
//# sourceMappingURL=index.js.map
