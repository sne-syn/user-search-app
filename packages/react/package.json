{
  "name": "@user-search-app/react",
  "version": "1.0.0",
  "main": "lib/index.js",
  "license": "MIT",
  "scripts": {
    "build": "rollup -c",
    "dev": "yarn build --watch",
    "test": "jest --verbose",
    "test:watch": "yarn test --watch"
  },
  "devDependencies": {
    "@babel/preset-env": "^7.18.10",
    "@babel/preset-react": "^7.18.6",
    "@babel/preset-typescript": "^7.18.6",
    "@rollup/plugin-image": "^2.1.1",
    "@testing-library/jest-dom": "^5.16.5",
    "@testing-library/react": "^13.3.0",
    "@types/jest": "^28.1.6",
    "@types/react": "^18.0.15",
    "jest": "^28.1.3",
    "jest-environment-jsdom": "^28.1.3",
    "react": "^18.2.0",
    "rollup": "^2.77.0",
    "rollup-plugin-img": "^1.1.0",
    "rollup-plugin-typescript2": "^0.32.1",
    "typescript": "^4.7.4"
  }
}
