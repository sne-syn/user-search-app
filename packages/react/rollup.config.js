import Ts from "rollup-plugin-typescript2";
import image from "@rollup/plugin-image";

export default {
  input: [
    "src/index.ts",
    "src/atoms/Avatar/index.ts",
    "src/atoms/Button/index.ts",
    "src/atoms/CardSubtitle/index.ts",
    "src/atoms/CardTitle/index.ts",
    "src/atoms/Icon/index.ts",
    "src/atoms/Input/index.ts",
    "src/atoms/Link/index.ts",
    "src/atoms/DefinitionList/index.ts",
    "src/atoms/DateInfo/index.ts",
    "src/atoms/Logo/index.ts",
    "src/atoms/TextBlock/index.ts",
    "src/atoms/ThemeToggler/index.ts",
    "src/atoms/VisuallyHidden/index.ts",
    "src/atoms/Wrapper/index.ts",
  ],
  output: {
    dir: "lib",
    format: "esm",
    sourcemap: true,
  },
  plugins: [
    Ts(),
    image({
      extensions: /\.(png|jpg|jpeg|gif|svg)$/,
      limit: 10000,
    }),
  ],
  preserveModules: true,
  external: ["react", "@ds.e/foundation"],
};
