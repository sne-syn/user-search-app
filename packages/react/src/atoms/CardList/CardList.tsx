import React, { FC } from "react";

interface CardListProps {
  children: React.ReactNode;
  style?: React.CSSProperties;
}
const CardList: FC<CardListProps> = ({ children, style }) => {
  return (
    <ul style={style} className="dse-card-list" data-testid="card-list">
      {children}
    </ul>
  );
};

export default CardList;
