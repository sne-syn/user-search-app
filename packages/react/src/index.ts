import Avatar from "./atoms/Avatar";
import Button from "./atoms/Button";
import CardSubtitle from "./atoms/CardSubtitle";
import CardTitle from "./atoms/CardTitle";
import Input from "./atoms/Input";
import ListContent from "./atoms/ListContent";
import ListIcon from "./atoms/ListIcon";
import Logo from "./atoms/Logo";
import DateInfo from "./atoms/DateInfo";
import DefinitionList from "./atoms/DefinitionList";
import ErrorMessage from "./atoms/ErrorMessage";
import TextBlock from "./atoms/TextBlock";
import ThemeToggler from "./atoms/ThemeToggler";
import VisuallyHidden from "./atoms/VisuallyHidden";
import Wrapper from "./atoms/Wrapper";

export {
  Avatar,
  Button,
  CardSubtitle,
  CardTitle,
  Input,
  ListContent,
  ListIcon,
  Logo,
  DateInfo,
  DefinitionList,
  ErrorMessage,
  TextBlock,
  ThemeToggler,
  VisuallyHidden,
  Wrapper,
};
