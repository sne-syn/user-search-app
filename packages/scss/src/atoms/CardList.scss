@import "foundation/all";

.dse-card-list {
  list-style: none;
  margin: map-get($spacing, "none");
  padding: map-get($spacing, "none");

  & > * {
    margin-bottom: map-get($spacing, "base");

    @include tablet {
      margin-bottom: map-get($spacing, "sm");
    }
  }
}
