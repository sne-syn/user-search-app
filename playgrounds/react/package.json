{
  "name": "@playgrounds/react",
  "version": "1.0.0",
  "main": "index.js",
  "license": "MIT",
  "scripts": {
    "dev": "parcel src/index.html -p 3000",
    "copy:assets": "cpx 'src/assets/**' 'dist/assets'"
  },
  "devDependencies": {
    "@types/react": "^18.0.15",
    "@types/react-dom": "^18.0.6",
    "cpx": "^1.5.0",
    "parcel-bundler": "^1.12.5",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "typescript": "^4.7.4"
  }
}
