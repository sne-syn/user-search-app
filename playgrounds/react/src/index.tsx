import React from "react";
import { createRoot } from "react-dom/client";

import { Button } from "@user-search-app/react";
import "@user-search-app/scss/lib/Button.css";
const container = document.querySelector("#root");
const root = createRoot(container!);
root.render(<Button label="Search"></Button>);
