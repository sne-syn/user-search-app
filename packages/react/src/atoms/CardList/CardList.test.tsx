import React from "react";
import CardList from "./CardList";
import { render, screen } from "@testing-library/react";

describe("CardList", () => {
  test("renders card list", () => {
    render(
      <CardList>
        <li>item</li>
      </CardList>
    );
    expect(screen.getByTestId("card-list")).toBeInTheDocument;
    expect(screen.getByTestId("card-list")).toMatchSnapshot();
  });

  test("renders card list with style provided", () => {
    render(
      <CardList style={{ width: "100px" }}>
        <li>item</li>
      </CardList>
    );
    expect(screen.getByTestId("card-list")).toHaveStyle("width: 100px");
    expect(screen.getByTestId("card-list")).toMatchSnapshot();
  });
});
