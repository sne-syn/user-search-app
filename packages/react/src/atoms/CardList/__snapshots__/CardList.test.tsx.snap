// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardList renders card list 1`] = `
<ul
  class="dse-card-list"
  data-testid="card-list"
>
  <li>
    item
  </li>
</ul>
`;

exports[`CardList renders card list with style provided 1`] = `
<ul
  class="dse-card-list"
  data-testid="card-list"
  style="width: 100px;"
>
  <li>
    item
  </li>
</ul>
`;
